src/analysis/utils/app-utils.ts

```diff
@@ -25,30 +36,31 @@ export const doesWorkspaceSupportCromwellAppForUser = (
   cloudProvider: CloudProvider,
   toolLabel: ToolLabel
 ): boolean => {
-  // deploy to prod happened around 9:45 AM EST
-  const workflowsPublicPreviewDate = new Date(Date.parse('Tue Mar 21 2023 10:00:00 GMT-0400')); // 10 AM EST
-
-  // note: for all environments, we want to show Cromwell app modal (both Settings and Open buttons) only for workspace creators
   return Utils.cond(
-    // for prod env, the ability to create and view Cromwell app for Azure workspaces is being introduced on March 21st, 2023 as part
-    // of the Workflows Public Preview. Azure workspaces before this would only have WDS app created when the workspace is created.
-    // Hence, for Azure workspaces that predate this launch date, we don't want to display the Cromwell app icon in Context bar and
-    // in Cloud Environment Modal we would like to display message asking users to create new workspace to use Cromwell app.
-    // Note: the date we are comparing against is March 21st since that's when the ability to create Cromwell app was merged into Prod.
-    [
-      toolLabel === appToolLabels.CROMWELL && cloudProvider === cloudProviderTypes.AZURE && getConfig().isProd,
-      () =>
-        new Date(workspaceInfo.createdDate).valueOf() > workflowsPublicPreviewDate.valueOf() &&
-        workspaceInfo.createdBy === getTerraUser()?.email,
-    ],
     [
       toolLabel === appToolLabels.CROMWELL && cloudProvider === cloudProviderTypes.AZURE,
       () => workspaceInfo.createdBy === getTerraUser()?.email,
     ],
+    [
+      toolLabel === appToolLabels.CROMWELL_RUNNER_APP && cloudProvider === cloudProviderTypes.AZURE,
+      () => isFeaturePreviewEnabled(ENABLE_AZURE_COLLABORATIVE_WORKFLOW_RUNNERS),
+    ],
+    [
+      toolLabel === appToolLabels.WORKFLOWS_APP && cloudProvider === cloudProviderTypes.AZURE,
+      () => isFeaturePreviewEnabled(ENABLE_AZURE_COLLABORATIVE_WORKFLOW_READERS),
+    ],
     [Utils.DEFAULT, () => true]
   );
 };
 
+export const getCurrentAppForUser = (appType: ToolLabel, apps: App[] | undefined): App | undefined =>
+  getCurrentApp(
+    appType,
+    apps?.filter(
+      (app) => app.accessScope !== appAccessScopes.USER_PRIVATE || app.auditInfo.creator === getTerraUser().email
```

Comment:
Hmm... What exactly does `appAccessScopes.USER_PRIVATE` represent? Leonardo already provides a label that we are using at the call site of this method to filter apps based on `app.auditInfo.creator === getTerraUser().email` via the `{ role: 'creator' }` api parameter specified here, so that should not be needed: https://github.com/DataBiosphere/terra-ui/blob/dev/src/pages/workspaces/workspace/WorkspaceContainer.js#L358

Should this access scope filter be added to leo as well?

Reply:
Hmm, so re-reading this, it seems like this function is not actually getting the app for the current user. Rather,  it is broadening 'currentApp' to be an app created by the current user OR a non-private app the user can see?

Reply:
As per [this Slack thread](https://broadinstitute.slack.com/archives/CA3NP1733/p1696971250202179), the role: creator param doesn't currently work with ListAppsV2. However, in this case we would expect to be able to view the shared apps even if we haven't created them (WDS, shared Workflows App). This is why I'm only filtering by creator if the app is a privately-scoped app.

Reply:
Though maybe this changed recently: https://github.com/DataBiosphere/leonardo/pull/3831

Reply:
Regardless, this seems like a very inadvisable thing to do in a util function - here we have a side affect (the current user) being used to filter data from the data client layer while filtering also exists at that layer via {role: creator} and is implemented in the backend. 

At the very least I think we should have a method that takes `Apps[] -> Apps[`] and does this filter. Then a function can apply the necessary filters to `apps`. I also think this function should not be called in the visual components at all, and should be something done in the `Apps.ts` data client layer while we lack a state provider. cc @msilva-broad  for insight on the best path forward while we mature the data client/state provider stack.

Ideally, I would like to move any `currentApp` logic to the backend service instead of layering more in the UI

A bit verbose, will discuss with IA team in stand-up tomorrow.

Reply:
I agree this would be better suited as backend functionality -- perhaps additional `role` values? There's three visibilities I can see us wanting to use in different cases:
1. Show me all of the apps in this workspace I can view, meaning any shared apps, all of my private apps, and (if I'm a workspace owner) all of other people's private apps
2. Show me all of the apps in this workspace I can access (phrasing?), meaning any shared apps and only my private apps (I believe listeners will reject owner from reaching private apps they didn't create, but they can still view it in the current listApps)
3. Show me all of the apps I created (may hide some shared apps)

1 is the existing functionality, and 3 was just added with the `role=creator` functionality in the previously mentioned Leo PR. But 2 is the one we would want in this case.

I'm also wondering how this would impact the workspace wrapper props. Currently we get the apps returned by case 3 from the workspace wrapper. But it feels strange to make 3 different calls to do said filtering (and it's also not as clear what the `apps` property contains)

Reply:
I agree that any filtering for "my app" vs "shared app" belongs in Leo rather than in the frontend. And I agree that (2) is what we probably want 9 times out of 10. The only exception I can think (maybe?) of would be during workspace deletion where you want to make sure everything's shut down before deleting the workspace.